docs/integration.md

```diff
@@ -0,0 +1,129 @@
+**transactionId** | The Hedera Transaction ID
+**wrappedAsset** | The corresponding wrappedToken contract address
+**receiver** | The address receiving the tokens
+**amount** | The desired minting amount
+**signatures** | The array of signatures from the members, authorising the operation
+
+### 1.4 Service fee distribution
+
+The main incentive for the Validators is the `service fee` paid by users. The fee is a percentage of the transferred amount, paid on the native asset. The Service fee is configurable property and determined by the validators.
+
+Fees are payed out from the Bridge account. Each Validator creates a Scheduled transaction and transfers the `service fee` amount from the Bridge account to the list of validators equally. Due to the nature of Scheduled Transactions, only one will be successfully executed, creating a scheduled Entity and all others will fail with `IDENTICAL_SCHEDULE_ALREADY_CREATED` error, and the transaction receipt will include the `ScheduleID` of the first submitted transaction. All validators, except the one that successfully created the Transaction execute `ScheduleSign` and once `n out of m` validators execute the Sign operation, the transfer of the fees will be executed.
+
+## 2. EVM-chain -> Hedera
+This functionality allows the user to transfer Wrapped HBAR or any supported by the bridge Wrapped Tokens from EVM-based chain to Hedera. 
+
+### 2.1 Supported wrapped tokens
+
+In order to get all the supported wrapped tokens by the bridge the user must do two things:
+
+ 1. Get the wrapped tokens count from the Router contract by calling the function `wrappedAssetsCount()`.
+ 2. Call the Router contract function `wrappedAssetAt(uint256  index)` for every value between 0 ... wrappedAssetsCount-1. Each time the function will return the address of the ERC20 wrapped token contract.
+
+If the user needs the name of the wrapped token he can retrieve it from public mapping in the Router contract by the ERC20 address:
+
+	mapping(address => bytes) public wrappedToNative;
+
+### 2.1 How it works
+Transfer from EVM-chain to Hedera is achieved by submitting a **burn operation** to the Router Contract. 
+There are two supported contract functions by which this can be done:
+
+ - `burn(amount: uint256, receiver: bytes, wrappedAsset: address)`
+
+Argument | Description
+---------- | ----------
+**amount** | The amount of wrapped tokens to be bridged
+**receiver** | The Hedera account to receive the wrapped tokens
+**wrappedAsset** | The corresponding wrapped asset contract address
+
+After the burn operation is completed a _burn_ event is fired which is captured by the validators. The event contains information about the burned amount and the receiver. After the validators capture the event they distribute the service fee and schedule a transaction to transfer the remaining amount to the receiving Hedera account.
+
+
+ - `burnWithPermit(wrappedAsset: address, receiver: bytes, amount: uint256, deadline: uint256, v, r ,s)`
```

Comment:
Maybe it would be better if we add information about the permit logic. Show a JS snippet on how to sign v,r,s and specify a `deadline`